### PR TITLE
Vps 062/popup screen

### PR DESCRIPTION
**Pull request by j-enn-y**
## Describe the issue

A clear and concise description of what the issue is.
#62 

## Describe the solution

A clear and concise description of what the solution is.

- Added red non-intrusive pop-up on bottom right corner which pops up when user has 30 seconds left
- Pop-up warns user "You have less than 30 seconds left!"
- 
image

## Risk

Potential risks that this PR might brings
- If user has placed anything under the pop-up, it will be hidden while the pop-up is shown.

## Definition of Done

- [ ] Code peer-reviewed
- [*] Wiki Documentation is written and up to date
- [ ] Unit tests written and passing
- [ ] Integration tests written and passing
- [ ] Continuous Integration build passing
- [ ] Acceptance criteria met
- [ ] Deployed to production environment

## Reviewed By

Who reviewed your PR - for commit history once merged

### CODE REVIEW

**Comment by Ryuna001** [comment]
- Leave 30 seconds warning in for 5 seconds
- Another pop up at 15 seconds, leave warning on for 5 seconds

**Comment by arnard76** [comment]
Hey @j-enn-y I converted this to a draft PR because it doesn't include Ryan's changes yet.

**Comment by j-enn-y** [comment]
image
image
image
image

Pop-up now lasts for 5 seconds and appears when there are 30 seconds and 15 seconds left in scene.

File: frontend/src/components/TimerComponent/PopUp.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const style = {
+  position: "absolute",
+  bottom: "0%",
+  right: "0%",
+  width: "20%",
+  bgcolor: "red",
+  border: "2px solid #000",
+  boxShadow: 24,
+  p: 4,
```
Comment (arnard76) [inline frontend/src/components/TimerComponent/PopUp.js:12]: Does this line do anything?

File: frontend/src/components/TimerComponent/index.js [hunk 2]
```diff
@@ -14,12 +15,20 @@ const CountdownTimer = ({ targetDate, sceneTime }) => {
...
 
   const [days, hours, minutes, seconds] = useCountdown(time);
-
   if (days + hours + minutes + seconds <= 0) {
-    return <Notifcation setTime={setTime} sceneTime={sceneTime} />;
+    return <Notification setTime={setTime} sceneTime={sceneTime} />;
   }
+  return (
+    <div>
+      {days + hours + minutes + seconds > 25 &&
```
Comment (arnard76) [inline frontend/src/components/TimerComponent/index.js:23]: you can try storing `days + hours + minutes + seconds` in a variable e.g. `time` or `totalTime` and use that in the conditional statements instead. It might make it easier to read 🙂

File: frontend/src/components/TimerComponent/PopUp.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+function PopUp({ seconds }) {
+  return (
+    <div>
+      {seconds > 15 && (
+        <Box sx={style}>
```
Comment (arnard76) [inline frontend/src/components/TimerComponent/PopUp.js:19]: Not sure the Box is needed. Can the styles be applied directly to the p tags?

File: frontend/src/components/TimerComponent/PopUp.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  width: "20%",
+  bgcolor: "red",
+  border: "2px solid #000",
+  boxShadow: 24,
+  p: 4,
+};
+
+function PopUp({ seconds }) {
+  return (
+    <div>
```
Comment (arnard76) [inline frontend/src/components/TimerComponent/PopUp.js:17]: Not sure this div is needed either. Could try replacing with a React fragment?
relevant docs: https://react.dev/reference/react/Fragment

**Comment by arnard76** [comment]
The popups are working ✅✅

I added some possible code improvements to your changes. Feel free to ignore if they don't make sense 🙂

File: frontend/src/components/TimerComponent/PopUp.js (hunk 3 above)
Comment (arnard76) [reply frontend/src/components/TimerComponent/PopUp.js:19]: This works
`<p style={style}>You have less than 30 seconds left!</p>`

but style has to be adjusted a bit. I don't think `p` is a built-in react style but `padding` seems to replace it.

File: frontend/src/components/TimerComponent/PopUp.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  backgroundColor: "red",
+  border: "2px solid #000",
+  padding: "40px",
+};
+
+function PopUp({ seconds }) {
+  return (
+    <>
+      {seconds > 15 && <p style={style}>You have less than 30 seconds left!</p>}
+
```
Comment (arnard76) [inline frontend/src/components/TimerComponent/PopUp.js:17]: A fun way to organize these two conditional statements is with the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator).

e.g. `{seconds > 15 ? 1st p element : 2nd p element}`

or maybe because the p element and the style is the same, just the text inside a single p element:
`<p style={style}>{seconds > 15 ? "less than 30s left" : "less than 15s left"}</p>`

Basically a nice shorthand for an if statement in javascript 🙂

**Review by djos192 — APPROVED** [review]
